bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandler.java

```diff
@@ -46,49 +42,66 @@
     Thing getThing();
 
     /**
-     * Handles a command for a given channel.
-     *
-     * @param channelUID
-     *            unique identifier of the channel to which the command was sent
-     * @param command
-     *            {@link Command}
+     * Initializes the thing handler, e.g. update thing status, allocate resources, transfer configuration.
+     * </p>
+     * This method is only called, if the {@link Thing} contains all required configuration parameters.
+     * </p>
+     * Only {@link Thing}s with status {@link ThingStatus#ONLINE} or {@link ThingStatus#OFFLINE} are
+     * considered as <i>initialized</i> by the framework. To achieve that, the status must be reported
+     * via {@link ThingHandlerCallback#statusUpdated(Thing, ThingStatusInfo)}.
+     * </p>
+     * The framework expects this method to be non-blocking and return quickly. For longer running initializations,
+     * the implementation has to take care of scheduling a separate job which must the status at the end.
      */
-    void handleCommand(ChannelUID channelUID, Command command);
+    void initialize();
 
     /**
-     * Handles a command for a given channel.
+     * Disposes the thing handler, e.g. deallocate resources.
+     * </p>
+     * The framework expects this method to be non-blocking and return quickly. For longer running disposals,
```

Comment:
I would not allow long running disposals. After all, this should NOT do any communication with the device, it should only clean up internal memory etc. It should be a valid assumption that the instance can be GC'ed immediately after dispose returns.